### DISCUSSION TITLE
Document previews

### DISCUSSION THREAD

**Discussion by gadfeldan** [protz/thunderbird-conversations#1569] [Ideas]
Thank you so much too for this wonderful addition .   I have two suggestions / requests.
1. Thunderbird-Conversations previews only pdf documents. Why doc documents cannot be previewed either Like in Gmail?

2. Is it possible to preview the documents before opening the email sent as in Gmail?                                                                                            Tank you very much in advance.   My Thunderbird version : 78.5.1 , My Thunderbird Conversation version: 3.2.8 , My Operating System: Windows 10 64 bit.

**Comment by Standard8** [comment]
I've converted this to a discussion, since that seems more appropriate here (and also as a bit of an experiment).

>     1. Thunderbird-Conversations previews only pdf documents. Why doc documents cannot be previewed either Like in Gmail?

There's probably a few reasons for this. Firstly, it hasn't been requested before - I'm not even sure Thunderbird itself has had a request for it.

It would also need us to have some sort of JavaScript library available for processing the document into a format that can be displayed. It might be that we'd also need several different libraries available to cope with most of the different document formats. If there wasn't a library we'd have to write one. Gmail have the advantage that their document processing is already online, so they can use all their own code to do a preview, we can't use that code unfortunately. I also wouldn't want to hand off documents to gmail to preview, as that would probably require some sort of license, but would also be a privacy concern.

This is something that we could potentially experiment with in the future and see if there's some compatible libraries that we could use.

>     2. Is it possible to preview the documents before opening the email sent as in Gmail?

It is not. I would say that this would be out of scope for Conversations, and would need to be addressed within Thunderbird itself or in another add-on.

**Reply by gadfeldan** [reply]
Thank you very much for the quick reply.
